### PR TITLE
Add section on type annotations

### PR DESCRIPTION
**Pull request by sschaub**

### CODE REVIEW

File: _sources/Functions/TypeAnnotations.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        return msg + msg
+
+This function is intended to duplicate a message; if called with the value 'Hello', it returns the value
+'HelloHello'. If called with other types of data, however, it will not work properly. (What will the
+function do if given an ``int`` or a ``float`` value?)
+
+Python allows you to indicate the intended type of the function parameters and the type of the function return value
+in a function definition using a special notation demonstrated in this example:
+
+.. sourcecode:: python
```
Comment (bnmnetp) [inline _sources/Functions/TypeAnnotations.rst:31]: You could make this a ..activecode:: directive as skulpt will cheerfully parse but ignore the annotations.  Which helps reinforce your message.

File: _sources/Functions/TypeAnnotations.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+type. A **type annotation** is an optional notation that specifies the type of a parameter or function result. It
+tells the programmer using the function what kind of data to pass to the function, and what kind of data to expect when
+the function returns a value.
+
+In the definition above, the annotation ``: str`` in ``msg: str`` indicates that the caller should pass a ``str`` value as an argument.
+The annotation  ``-> str`` indicates that the function will produce a ``str`` result. 
+
+Here are some more examples of functions with type annotations:
+
+.. sourcecode:: python
```
Comment (bnmnetp) [inline _sources/Functions/TypeAnnotations.rst:48]: same as above

File: _sources/Functions/TypeAnnotations.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+to check that the values passed to a function are the expected types, or cause the returned value to be converted to the
+expected type. For example, if the function ``add`` in the example above is called like this::
+
+    result = add('5', '15')
+
+the function will receive two string values, concatenate them, and return the resulting string '515'. The ``int``
+annotations are completely ignored by the Python interpreter. Think of type annotations as a kind of function
+documentation, and remember that they have no effect on the program's behavior.
+
+Type annotations are an optional aspect of documenting functions. Still, type annotations are an important tool to increase
```
Comment (bnmnetp) [inline _sources/Functions/TypeAnnotations.rst:77]: would you want to mention mypy or similar to say that there are tools to analyze your code that will flag problems?

How about adding a couple of check your understanding questions?  Given an annotated function called "correctly" what would be the result?  called "incorrectly" what would be the result?

Comment (bnmnetp) [reply _sources/Functions/TypeAnnotations.rst:77]: The thought process I usually have in mind is to write questions, as our research shows that a lot of times students will look at the questions and try to answer them before they read.  If they don't know the answer then they'll go back and read to find it.

Comment (sschaub) [reply _sources/Functions/TypeAnnotations.rst:89]: Done!

**Pull request merged**